File: telegram-bot-spring-boot-starter/src/main/java/ru/tggc/telegrambotspringbootstarter/TelegramLifecycle.java
package ru.tggc.telegrambotspringbootstarter;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.SmartLifecycle;
import ru.tggc.telegrambotspringbootstarter.runner.TelegramBotRunner;

@Slf4j
@RequiredArgsConstructor
public class TelegramLifecycle implements SmartLifecycle {
    private boolean running = false;

    private final TelegramBotRunner telegramBotRunner;

    @Override
    public void start() {
        telegramBotRunner.start();
        running = true;
    }

    @Override
    public void stop() {
        log.info("Stopping Telegram bot runner");
        running = false;
    }

    @Override
    public boolean isRunning() {
        return running;
    }
}

File: telegram-bot-spring-boot-starter/src/main/kotlin/ru/tggc/telegrambotspringbootstarter/TelegramLifecycle.kt
package ru.tggc.telegrambotspringbootstarter

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.context.SmartLifecycle
import ru.tggc.telegrambotspringbootstarter.runner.TelegramBotRunner

class TelegramLifecycle(private val runner: TelegramBotRunner) : SmartLifecycle {
    private val log = KotlinLogging.logger {}
    private var running = false

    override fun start() {
        runner.start()
        running = true
    }

    override fun stop() {
        log.info { "Stopping Telegram bot runner" }
        running = false
    }

    override fun isRunning(): Boolean = running

}
